server/src/main/java/org/opensearch/index/engine/NRTReplicationEngine.java

```diff
@@ -85,6 +85,7 @@ public NRTReplicationEngine(EngineConfig engineConfig) {
 
     public synchronized void updateSegments(final SegmentInfos infos, long seqNo) throws IOException {
         // Update the current infos reference on the Engine's reader.
+        assert engineConfig.isReadOnlyReplica() : "Only replicas should update Infos";
```

Comment:
nit - This assert isn't really necessary, we assert the engine type in IndexShard & NRTReplicationEngine would never be wired up if it were not a readOnlyReplica.